rtmpsuck.c

```diff
@@ -776,9 +774,23 @@ TFTYPE doServe(void *arg)	// server socket and state (our listening socket)
     {
       if (!RTMPPacket_IsReady(&ps))
         continue;
-      ServePacket(server, 0, &ps);
+      /* Client can set chunk_size before the connect command */
+      if (ps.m_packetType == RTMP_PACKET_TYPE_CHUNK_SIZE)
+        {
+          if (ps.m_nBodySize >= 4)
+            {
+              server->rs.m_inChunkSize = AMF_DecodeInt32(ps.m_body);
+              RTMP_Log(RTMP_LOGDEBUG, "%s, client: chunk size change to %d", __FUNCTION__,
+                       server->rs.m_inChunkSize);
+            }
+        }
+      else
+        {
+          ServePacket(server, 0, &ps);
+          connect_cmd_processed = TRUE;
```

Comment:
But since it's set after the first call to `ServePacket()`, it's not tracking the state of the connect command, it's just tracking that `ServePacket()` was called.

Essentially, this is happening:

``` c
int connect_cmd_processed = FALSE;

while (...) {
	if (!RTMPPacket_IsReady(&ps))
		continue;
	ServePacket(server, 0, &ps);
	connect_cmd_processed = TRUE;
	RTMPPacket_Free(&ps);
	if (connect_cmd_processed && RTMP_IsConnected(&server->rc))
		break;
}
```